src/Json.Schema.ToDotNet.UnitTests/Hints/ClassNameHintTests.cs

```diff
@@ -269,7 +269,208 @@ public int GetHashCode(FileData obj)
         }
     }
 }"
-)
+                ),
+
+            new TestCase(
+                "Renamed class has a base class",
+@"{
+  ""type"": ""object"",
+  ""properties"": {
+    ""file"": {
+      ""$ref"": ""#/definitions/file""
+    }
+  },
+  ""definitions"": {
+    ""file"": {
+      ""type"": ""object"",
+      ""properties"": {
+        ""path"": {
+          ""type"": ""string""
+        }
+      }
+    }
+  }
+}",
+
+    "FileData",
+
+@"{
+  ""file"": [
+    {
+      ""kind"": ""ClassNameHint"",
+      ""arguments"": {
+        ""className"": ""FileData""
+      }
+    }
+  ],
+  ""fileData"": [
+    {
+      ""kind"": ""BaseTypeHint"",
+      ""arguments"": {
+        ""baseTypeNames"": [
+          ""PropertyBagHolder""
+        ]
+      }
+    }
```

Comment:
The `BaseTypeHint` says that the renamed class has a base class `PropertyBagHolder`.